BraintreeCore/src/main/java/com/braintreepayments/api/core/ConfigurationCache.kt

```diff
@@ -1,37 +1,45 @@
 package com.braintreepayments.api.core
 
 import android.content.Context
+import android.util.Base64
 import com.braintreepayments.api.sharedutils.BraintreeSharedPreferences
+import org.json.JSONException
 import java.util.concurrent.TimeUnit
 
 internal class ConfigurationCache(
     private val sharedPreferences: BraintreeSharedPreferences
 ) {
 
-    fun getConfiguration(cacheKey: String): String? {
-        return getConfiguration(cacheKey, System.currentTimeMillis())
-    }
-
-    fun getConfiguration(cacheKey: String, currentTimeMillis: Long): String? {
+    fun getConfiguration(
+        authorization: Authorization,
+        configUrl: String,
+        currentTimeMillis: Long = System.currentTimeMillis()
```

Comment:
What are your thoughts on injecting a class that can handle getting `System.currentTimeMillis()`? This way we could inject that class into `ConfigurationCache`'s primary constructor. We could then remove the parameter from both `getConfiguration` and `putConfiguration`.

Reply:
Coming up with a name is the hardest part 😂 What about `Time.getCurrentTime()`? I could see it also handling anywhere where we deal with timestamps (possibly timestamps in analytic events).

This isn't a current pattern, but thinking in MVVM, I could see it being it's own use case: `GetCurrentTimeUseCase`.

Reply:
> Coming up with a name is the hardest part 😂 

Forreal! I'm about to get this as a tattoo lol 😭. I like Time! I'm gonna update the PR. It does clean up both the src and test files which is awesome 😮‍💨